packages/@yoda/network/index.js

```diff
@@ -0,0 +1,141 @@
+'use strict'
+
+/**
+ * @module @yoda/network
+ * @description Provides classes to manage network functions on the device.
+ */
+
+var EventEmitter = require('events').EventEmitter
+var FloraComp = require('@yoda/flora/comp')
+
+class NetworkAgent extends EventEmitter {
+  constructor () {
+    super()
+
+    this._remoteCallTarget = 'net_manager'
+    this._remoteCallCommand = 'network.command'
+    this._remoteCallTimeout = 60 * 1000
+
+    this._flora = new FloraComp('@network')
+    this._flora.init()
+  }
+
+  subscribe () {
+    this._flora.subscribe('network.status', (caps, type) => {
+      var msg = JSON.parse(caps[0])
+
+      if (msg.network) {
+        this.emit('network.status', msg.network)
+      } else if (msg.wifi) {
+        this.emit('wifi.status', msg.wifi)
+      } else if (msg.ethernet) {
+        this.emit('ethernet.status', msg.ethernet)
+      } else if (msg.modem) {
+        this.emit('modem.status', msg.modem)
+      }
+    })
+  }
+
+  _remoteCall (device, command, params) {
+    var data = {
+      device: device,
+      command: command
+    }
+    if (params) { data.params = params }
+
+    return this._flora.call(
+      this._remoteCallCommand,
+      [JSON.stringify(data)],
+      this._remoteCallTarget,
+      this._remoteCallTimeout
+    ).then(reply => {
+      return Promise.resolve(JSON.parse(reply.msg[0]))
```

Comment:
If you have in the function body of `then(fn)`, `Promise.resolve` or `Promise.reject` are not needed:

```suggestion
      return JSON.parse(reply.msg[0])
```